activerecord/test/cases/persistence_test.rb

```diff
@@ -120,6 +120,15 @@ def test_increment_attribute_by
     assert_equal 59, accounts(:signals37, :reload).credit_limit
   end
 
+  def test_increment_concurency
+    a1 = accounts(:signals37)
+    initial_credit = a1.credit_limit
+    a2 = Account.find(accounts(:signals37).id)
+    a1.increment!(:credit_limit)
+    a2.increment!(:credit_limit)
+    assert_equal initial_credit + 2, a1.reload.credit_limit
+  end
+
```

Comment:
@jeremy Maybe they fix some bugs when counter cache is only updated in DB but not on instance, but it is hard to assume where it happens and if it really happens. We have no identity map and we can not be sure we updated all instances in memory with a proper counter value.

So I would consider all these changes a refactoring that just reuses code. 

Renamed test name to be more precise about use case.